File: tests/fixtures/type-infer-unannotated.input.ts
class Greeter {
    name: string;
    constructor(name: string) {
        this.name = name;
    }
    greet(msg: string): string {
        return `${this.name}: ${msg}`;
    }
}

function main(): void {
    // const without type annotation — should infer Greeter
    const g = new Greeter("Alice");
    console.log(g.greet("hello"));

    // let without type annotation — should also infer
    let h = new Greeter("Bob");
    console.log(h.greet("world"));
}

File: tests/fixtures/type-infer-unannotated.input.rs
#[derive(Debug, Clone, PartialEq)]
struct Greeter {
    name: String,
}

impl Greeter {
    fn new(name: String) -> Self {
        Self { name }
    }

    fn greet(&self, msg: String) -> String {
        format!("{}: {}", self.name, msg)
    }
}

fn main() {
    let g = Greeter::new("Alice".to_string());
    println!("{}", g.greet("hello"));
    let mut h = Greeter::new("Bob".to_string());
    println!("{}", h.greet("world"));
}
